examples/server.cc

```diff
@@ -1228,6 +1235,25 @@ int Server::init(int fd) {
   return 0;
 }
 
+int Server::on_write() {
+  int rv = NETWORK_ERR_OK;
+
+  for (auto it = std::begin(handlers_);
+       it != std::end(handlers_); ++it) {
+    Handler *h = it->second.get();
+
+    auto hrv = h->on_write();
+    if (hrv == NETWORK_ERR_SEND_NON_FATAL) {
+      rv = NETWORK_ERR_SEND_NON_FATAL;
+    }
+    if (hrv == NETWORK_ERR_SEND_FATAL) {
+      return NETWORK_ERR_SEND_FATAL;
```

Comment:
Personally, I think that switch ... case is more readable here.

Reply:
I agree with all of the above comments and will address them soon.  Thanks.